_LowLevelCode/CPP/cpp_communicator/cpp_communicator.cpp

```diff
@@ -44,16 +100,21 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
     }
 
     //* Check and parce input  arguments. */
-    char *data_buffer(nullptr);
+    uint8_t * data_buffer(nullptr);
+
+    std::vector<int32_t> data_addresses;
+    std::vector<int32_t> data_tag;
+    bool is_synchroneous(false);
 
-    int data_address(0), data_tag(0);
+    AdditinalParamHolder AddPar;
     size_t n_workers;
     size_t nbytes_to_transfer;
     input_types work_type;
 
 
     class_handle<MPI_wrapper>* pCommunicatorHolder = parse_inputs(nlhs, nrhs, prhs,
-        work_type, data_address, data_tag, nbytes_to_transfer, data_buffer);
+        work_type, data_addresses, data_tag,is_synchroneous,
+        data_buffer, nbytes_to_transfer, AddPar);
 
     // avoid problem with multiple finalization
     if (pCommunicatorHolder == nullptr) { // this can happen only if close_mpi is selected and the framework had been already finalized
```

Comment:
missing brackets around `if (nlhs>0) { body }`

Reply:
let me leave it as it is until we decide on bracket policy